velero-plugin-for-microsoft-azure/volume_snapshotter.go

```diff
@@ -87,6 +87,7 @@ func newVolumeSnapshotter(logger logrus.FieldLogger) *VolumeSnapshotter {
 
 func (b *VolumeSnapshotter) Init(config map[string]string) error {
 	if err := veleroplugin.ValidateVolumeSnapshotterConfigKeys(config,
+		vslConfigKeyActiveDirectoryAuthorityURI,
```

Comment:
@kaovilai 

Please take a look here: https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/blob/7cc3a717566a05f4e8c7eeaf01031c77b5316835/volumesnapshotlocation.md?plain=1#L40

What happens is that when this value is specified it it will error out on the following line during its check.

Also the Readme also has information for backup locations:
https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/blob/main/README.md#create-backup-storage-location

Reply:
I get the validation part. I'm trying to understand which part of the code consumes this config.

Reply:
@kaovilai 

Any time config is called.

https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/blob/7cc3a717566a05f4e8c7eeaf01031c77b5316835/velero-plugin-for-microsoft-azure/volume_snapshotter.go#L100C22-L158

Its within the azure go library, if you are looking for the specifics ?

https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/blob/7cc3a717566a05f4e8c7eeaf01031c77b5316835/velero-plugin-for-microsoft-azure/volume_snapshotter.go#L139

Follow that chain. 

https://github.com/vmware-tanzu/velero/blob/6ac38cde8556f0fb338f1b00cd64746790b6c73c/pkg/util/azure/util.go#L82

https://github.com/vmware-tanzu/velero/blob/6ac38cde8556f0fb338f1b00cd64746790b6c73c/pkg/util/azure/util.go#L149C5-L149C33

Reply:
ok I see. Thanks.